### PR TITLE
feat: Add LSP support for the DOT/Graphviz lang

### PR DESCRIPTION
**Pull request by abougouffa**
This PR adds a new LSP client for the DOT language used in [Graphviz](https://graphviz.org/) via the [dot-language-server](https://github.com/nikeee/dot-language-server)

### CODE REVIEW

**Comment by kiennq** [comment]
Please add some docs for it as well.

**Comment by abougouffa** [comment]
@kiennq I've just added it to `docs/lsp-clinets.json`.

I didn't fine another place where to place relevant documentation, the LS is quite straight forward, there is no special parameters or instructions to expand in a separate documentation section.

Is there any other place where to add the client's documentation?

File: clients/lsp-dot.el [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+(defun lsp-dot--dot-ls-activate-p (filename &optional _)
+  "Check if the dot-ls lsp server should be enabled based on FILENAME."
+  (or (string-match-p "\\.dot\\|\\.gv\\'" filename)
+      (derived-mode-p 'graphviz-dot-mode)))
+
+(lsp-register-client
+ (make-lsp-client
+  :new-connection (lsp-stdio-connection #'lsp-dot--dot-ls-server-command)
+  :priority -1
+  :activation-fn #'lsp-dot--dot-ls-activate-p
```
[the remark this answers is not in the document]
Comment (abougouffa) [reply clients/lsp-dot.el:54]: Well, it can be! I will modify it

**Review by yyoncho — APPROVED** [review]

**Comment by jcs090218** [comment]
Update changelog?

**Comment by abougouffa** [comment]
@jcs090218 Done!

**Review by jcs090218 — APPROVED** [review]

**Review by kiennq — APPROVED** [review]

**Pull request merged**